### PR TITLE
UCT/IB/MLX5: Disable ODP on DEVX

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11806/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 15-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/8e48bbf48ae82bacd4277f815cd06c87599b8042/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/d515990599042bfff93e75552e5ce9a87ca2e64a/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/97f40de0b2ab6642eafde474fce905461f37615d/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/c630ae626d1cd79a979bfd77115827d7f8682668/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/7f1ac0bc62d9ef32ca533f876a8fe66e7e178ee9/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/cd4f46abe79c7a4ccfd2cb1d81cf129f2bf92eb3/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/c2276e829ecae2b98e6865f925b841f7fd34c5c6/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/f6a927156340dee17f3586fcf0f68b12264861ca/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/037f1dcb9b75d7e9569261a1395264d548f0f06e/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/ae7cf98ebd3bae67486089050dc956fa9466cc09/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/1d68a2df0dac2c4b0e4eb188b13fa6bf3ed591bc/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/e172b2957d03b72a0d8760cae2b59e66cad8625b/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/55770481db7b491a1a603bfa59f31dbcd4aed978/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/9b6ecaa2475cbaa0ed801897c0441b2e75842b53/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/e4e9b7733b3d23c9c509156a9891c6cfc4421838/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/27c4bd57f59deee1025c3fc909b885894ec36d43/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/ea685428eea4097d5a23d2f0a4f2909fc393c255/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/0de1a796d638b8e7c26cca0d2d85ce75bc50cf49/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/340a9b1668ffbfc566dbd5d9f6d281471477239e/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/aa771cbc8dfbeed4ffb4cb769ec25a53cf263b80/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/63806134e7792f8d33e1d049c99a21b9736d218d/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/f711972b2d7733841dd484559b20ed7d9266dabc/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/0fe935e2fe11df899fb7bdae979db28b988b3bfa/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/676076ab983cfbc9b12f9041c2ded20d2f61af82/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/f48f3b98b3259b3eaa358c23bf5c260699b0f727/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/4a2273b5b44b9d5a3cd0b12c34fb1c2164422a3e/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/2454dad47c7f11643f82a970163d2cc26bfca918/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/8e0ecec5f859eb086d63ab268d7b9a5a3dc7ef41/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a2e3c844b0f6cf070b9e22ed38897788/raw/c724363bbf5e961258ee5f595d3e7a6d4c13a8ea/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by brminich** [comment]
perf check in ud/test_ucp_perf.envelope/0

bot:bgate:retest

File: src/uct/ib/mlx5/dv/ib_mlx5dv_md.c [hunk 1]
```diff
@@ -404,10 +405,12 @@ static ucs_status_t uct_ib_mlx5_devx_md_open(struct ibv_device *ibv_device,
         md->flags |= UCT_IB_MLX5_MD_FLAG_KSM;
     }
 
+#if HAVE_DEVX_ODP
```
Comment (yosefe) [inline src/uct/ib/mlx5/dv/ib_mlx5dv_md.c]: can we disable this in runtime bu DEVX flag
because user can compile with DEVX but turn it off

Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/dv/ib_mlx5dv_md.c]: Yes, we do. It is DEVX flow

Comment (yosefe) [reply src/uct/ib/mlx5/dv/ib_mlx5dv_md.c]: so if HAVE_DEVX_ODP is like disabling it for now?
maybe add some comment here?

Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/dv/ib_mlx5dv_md.c]: Yeah.
BTW shouldn't we open RM - to close the gap?
Then could add /* TODO pending resolution of issue NN ...

Comment (yosefe) [reply src/uct/ib/mlx5/dv/ib_mlx5dv_md.c]: Yep, can you pls open?

**Comment by yosefe** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11825/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 17-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/a80872908d5114022e46c7701a6fac7e939411cc/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/4820ad5578a8d71e7a0358d883d6a9d7f098b9ea/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/cb4dcbfae80efa71e197771318d152760525e501/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/53a87f86f2be6f67e5ff27d02624eb7dd6a69baa/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/573ba628c0df5d1d7d7cf17df38b1db06f6eafd1/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/e5756a4207b0743e5af3a8a7c563609ab65a3553/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/152297ae2d6cd135c929da88843d5a963aa53e85/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/5a7e48c8de1274662ae138114e4d5c0ecd7eab1d/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/aad8feb06388090d2f72010bbb4a3fb07e1cf10f/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/e97465bf36281cf243e04ad876033c1fb7242157/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/e841aa90d69fbf37d0cd5f9e7c9a26e3fa938435/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/e00c50dae702eca2ea3fab43498d6a375b81fdc6/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/d13b094cf58ae15d5a32107b4ec3e38c693d651f/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/769a02a84231bc2c9873ab57aef52a9dbd28301e/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/930930c29d7ccd01e1f4935188b5fbc3ee3f1aba/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/a664bfdc111388468447da46d07c606f871ffa2e/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/84fabeccbe9ac8684aa95f1fdb36f401c9bd41ca/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/67e56971c047dcfa053cb80f365f7fd05682f5d8/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/5d2c4b1a8f32ae94a944fe1ef88d842e62b66952/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/42dd94ca72053904c8110f5b9e69f0451f902620/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/cc3607f67d37fb49e8bb3f85fc69a984603cf8f4/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/e90adba2a8ce7a08e243b24f84b28d0994e82399/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/67d1b727955bf8834a864314c8b60b60e01922cc/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/cde9612ee071178cd28657b026ef0c5c3c4783be/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/a98c45dd4ccad62cd2fbaf728d81fcc7f343a56b/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/63878517f6515423845fd303070cb943393555f7/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/8e9426ddda2fdcbf962cdc469736e302db88ea21/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/ea3910f155abe024ebe1be484921412af59d9a97/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/564616e95ff234b413a7857ef55fae6a/raw/5fe3a1275a60fc5de158975fc6eb47632f882e1b/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
bot:bgate:retest

**Pull request merged**